### PR TITLE
Frontend refactors 

### PR DESCRIPTION
**Pull request by sei-aderr**
Addresses all points in #359

### CODE REVIEW

File: mlte/frontend/nuxt-app/pages/index.vue [hunk 1]
```diff
@@ -305,11 +318,12 @@ async function selectModel(modelName: string, initialPageLoad: boolean) {
     selectedVersion.value = "";
   }
   if (modelName === "") {
+    versionOptions.value = [];
     clearArtifacts();
     return;
   }
 
-  await useFetch("http://localhost:8080/api/model/" + modelName + "/version", {
+  await $fetch("http://localhost:8080/api/model/" + modelName + "/version", {
```
Comment (sebastian-echeverria) [inline mlte/frontend/nuxt-app/pages/index.vue:326]: Could you reduce the duplication of the hardcoded localhost:8080 path in some way? Such as by having a base fetch function that fetchArtifact and these other fetches call?

**Review by sebastian-echeverria — CHANGES_REQUESTED** [review]
Changes look ok, but take a look at my comment below to see if we could add that as an internal improvement as well. Having that hardcoded server URL is a pain, but having it duplicated all over the code is worse. Might be a good time to reduce that duplication.

**Review by sebastian-echeverria — APPROVED** [review]
Nice! This actually addresses #252. Sending this to Kate for her review.

**Comment by sei-aderr** [comment]
I think it is close to addressing that, but not 100% of the way there. Kyle's idea with that was to be able to feed in the backend URL with a CLI command or something so that it was configurable from a user perspective which one to point it at. I could set up a prod/dev endpoint or something similar but I am not sure how I would go about providing a way for the user to edit that value during the use of the package.

**Comment by sebastian-echeverria** [comment]
Ok, fair enough. We can deal with that improvement later.

**Review by krmaffey — APPROVED** [review]
This looks great, thanks both!

**Pull request merged**